pkg/webhook/merge_on_review_approval_and_status_checks.go

```diff
@@ -40,27 +40,41 @@ func (h *autoMerger) HandleEvent(w http.ResponseWriter, payload interface{}, ghC
 		return h.handlePullRequestEvent(w, event, ghClientFunc)
 	case *github.StatusEvent:
 		return h.handleStatusEvent(w, event, ghClientFunc)
+	case *github.PullRequestReviewEvent:
+		return h.handlePullRequestReviewEvent(w, event, ghClientFunc)
 	default:
 		return nil
 	}
 }
 
+func (h *autoMerger) handlePullRequestReviewEvent(w http.ResponseWriter, event *github.PullRequestReviewEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
```

Comment:
So if I understand you right you want to merge in case of two use case:

* Someone labels directly the PR as "approved" without doing an explicit approcal step.
* Someone approves vie "Approved" step in the review workflow.

I wonder whether the flow here to check the approval directly is necessary as an approval already triggers the generation of the label which in turn triggers the merge (case 1). Is it because its more direct (because it contains only 1 step instead of 2) ? Just an other safeguard, or ?